win32ss/user/ntuser/kbdlayout.c

```diff
@@ -583,12 +583,10 @@ co_UserActivateKbl(PTHREADINFO pti, PKL pKl, UINT Flags)
     PWND pWnd;
 
     pklPrev = pti->KeyboardLayout;
-    if (pklPrev)
-        UserDereferenceObject(pklPrev);
 
-    pti->KeyboardLayout = pKl;
+    UserAssignmentLock((PVOID*)&(pti->KeyboardLayout), pKl);
```

Comment:
No `UserAssignmentUnlock` call for the previous value of `pti->KeyboardLayout` ?

Reply:
The previous value will released by assignment-lock if not same nor null.